resources/com.sap.piper/pipeline/stageDefaults.yml

```diff
@@ -70,6 +70,9 @@ stages:
       neoDeploy:
         configKeys:
           - 'neo/account'
+      tmsUpload:
+        configKeys:
+          - nodeName
```

Comment:
```suggestion
          - 'nodeName'
          - 'mtaPath'
          - 'credentialsId'
```

Reply:
Not sure about the suggestion. For all other cases there is only some lightweight check for the "most important" property (... whatever this is) and tried to mimic that.

Reply:
The "advantage" I see is:  in case the mtaPath or the credentialsId is missing TMS will fail. With the suggestion TMS will be skipped. On the other hand TMS will be skipped when no `nodeName` has been set. Another - maybe cleaner - approach would be to work with boolean flags, but others might see that also as some disadvantage.

Reply:
In general, above 3 parameters are mandatory to executed `tmsUpload` [step](https://github.com/SAP/jenkins-library/blob/master/vars/tmsUpload.groovy#L84). Missing of either of them does not run `tmsUpload`. To make it consistent with other files, I have added this suggestion.

Reply:
> In general, above 3 parameters are mandatory

I know.

> Missing of either of them does not run

Missing either of them will cause the tmpUpload to fail. Having all three parameters denoted here will skip the tmpUpload. Having only one denoted here will trigger the tmsUpload and let it fail in case the others are missing. That is IMO a clearer appraoch. Would be even better IMO - as already outlined above - to have a boolean flag or to check for the tms node in the config itself. But I simply tried to mimic what I found.

> To make it consistent with other files

Which other files. I tried to make this change consistent with the other parts in this file.

Reply:
> Having all three parameters denoted here will skip the tmpUpload. Having only one denoted here will trigger the tmsUpload and let it fail in case the others are missing. 
>
Ok, I didn't knew this that the whole tmsUpload step will be skipped, if we add 3 parameters here.
>

> Which other files. I tried to make this change consistent with the other parts in this file.

I mean the [cloudSdkStageDefaults ](https://github.com/SAP/jenkins-library/blob/master/resources/com.sap.piper/pipeline/cloudSdkStageDefaults.yml#L95)file.

Reply:
Also in that file checking all parameters instead of a "significant" one seems to be more the exception that the norm.